hoodie-client/pom.xml

```diff
@@ -28,6 +28,15 @@
     <build>
         <plugins>
             <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.18.1</version>
+                <configuration>
+                    <reuseForks>false</reuseForks>
+                    <forkCount>1</forkCount>
+                </configuration>
+            </plugin>
+            <plugin>
```

Comment:
After making changes to TestHoodieClient to use DFS instead of LocalFS, some other tests under hoodie-common package started to fail. After digging into the problem, I found that JUnit runs all tests under a module in the same JVM. Some classes (for example HoodieTestUtils) are static and hold a reference to FileSystem throughout the lifecycle of the JVM. The problem is if there are tests in the same module that use LocalFS and DFS, depending on which FS gets initialized first, that FS is used throughout causing other tests to fail. I could think of a few approaches to solve this issues : 
1) This plugin allows to teardown and restart the JVM for every TestClass under that module. This sounds like an overhead although I haven't seen tests taking longer than usual.
2) Fix all such static classes to either take filesystem as an argument in their methods, it is cumbersome to change all and we would have to follow such a pattern for every static class using a filesystem.
3) Re-set the filesystem object in the Test, for example AnyClass.fs = <whatever fs we want>